### PR TITLE
SCB-315 Config Center module can't get config by DynamicPropertyFacto…

### PR DESCRIPTION
**Pull request by jeho0815**
…ry.getInstance().getStringProperty, so has to use ConcurrentCompositeConfiguration

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/SCB) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[SCB-XXX] Fixes bug in ApproximateQuantiles`, where you replace `SCB-XXX` with the appropriate JIRA issue.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean install` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

File: foundations/foundation-vertx/src/main/java/org/apache/servicecomb/foundation/vertx/AddressResolverConfig.java [hunk 1]
```diff
@@ -32,58 +33,89 @@
...
+            AddressResolverOptions.DEFAULT_ROTATE_SERVERS,
             "addressResolver." + tag + ".rotateServers",
             "addressResolver.rotateServers"));
     return addressResolverOptions;
   }
 
-  private static List<String> getStringListProperty(List<String> defaultValue, String... keys) {
+  private static List<String> getStringListProperty(ConcurrentCompositeConfiguration configSource,
+      List<String> defaultValue, String... keys) {
     String property = null;
```
Comment (wujimin) [inline foundations/foundation-vertx/src/main/java/org/apache/servicecomb/foundation/vertx/AddressResolverConfig.java]: if configSource is null, then assign from DynamicPropertyFactory.getBackingConfigurationSource()
then return Arrays.asList(configSource.getStringArray(key))

Comment (jeho0815) [reply foundations/foundation-vertx/src/main/java/org/apache/servicecomb/foundation/vertx/AddressResolverConfig.java]: ok

File: foundations/foundation-vertx/src/main/java/org/apache/servicecomb/foundation/vertx/AddressResolverConfig.java [hunk 2]
```diff
@@ -95,33 +127,55 @@ public static AddressResolverOptions getAddressResover(String tag) {
     }
   }
 
-  private static int getIntProperty(int defaultValue, String... keys) {
+  private static int getIntProperty(ConcurrentCompositeConfiguration configSource, int defaultValue, String... keys) {
```
Comment (wujimin) [inline foundations/foundation-vertx/src/main/java/org/apache/servicecomb/foundation/vertx/AddressResolverConfig.java]: we can make the logics simpler

File: foundations/foundation-vertx/src/main/java/org/apache/servicecomb/foundation/vertx/AddressResolverConfig.java [hunk 3]
```diff
@@ -32,96 +34,113 @@
...
+  private static int getIntProperty(Configuration configSource, int defaultValue, String... keys) {
+    if (configSource == null) {
+      configSource = (Configuration) DynamicPropertyFactory.getBackingConfigurationSource();
+    }
     for (String key : keys) {
-      property = DynamicPropertyFactory.getInstance().getIntProperty(key, -1).get();
-      if (property > 0) {
-        break;
+      Integer val = configSource.getInteger(key, null);
+      if (val != null && val > 0) {
```
Comment (wujimin) [inline foundations/foundation-vertx/src/main/java/org/apache/servicecomb/foundation/vertx/AddressResolverConfig.java]: why a value <=0 is invalid?

Comment (jeho0815) [reply foundations/foundation-vertx/src/main/java/org/apache/servicecomb/foundation/vertx/AddressResolverConfig.java]: all integer parameters must large than zero

Comment (wujimin) [reply foundations/foundation-vertx/src/main/java/org/apache/servicecomb/foundation/vertx/AddressResolverConfig.java]: 1.if there is some property have no this limit
and developers see there is a getIntProperty method, invoke it directly......

2.users make a wrong configuration, here ignore the wrong value, and use default value
   but this behavior is silence
   nobody know what happened.

**Pull request merged**